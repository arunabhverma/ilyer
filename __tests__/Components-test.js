import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Box from '../src/components/box';
import Card from '../src/components/card';
import CustomButton from '../src/components/customButton';
import Image from '../src/components/image';
import Text from '../src/components/text';

const box = renderer.create(<Box />).toJSON();
const card = renderer.create(<Card />).toJSON();
const customButton = renderer.create(<CustomButton />).toJSON();
const image = renderer.create(<Image />).toJSON();
const text = renderer.create(<Text />).toJSON();

test('snapshot', () => {
  expect(box).toMatchSnapshot();
  expect(card).toMatchSnapshot();
  expect(customButton).toMatchSnapshot();
  expect(image).toMatchSnapshot();
  expect(text).toMatchSnapshot();
})


jest.mock('react-native-text-input-mask', () => ({
  default: jest.fn(),
}))

jest.mock('react-native-vector-icons/Ionicons', () => ({
  default: jest.fn(),
}))
