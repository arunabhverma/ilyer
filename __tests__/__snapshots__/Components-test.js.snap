// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View
  style={
    Array [
      Object {},
    ]
  }
/>
`;

exports[`snapshot 2`] = `
<View
  style={
    Array [
      Object {
        "borderBottomWidth": 0,
        "borderColor": "rgba(56, 146, 250, 0.1)",
        "borderTopWidth": 0,
        "borderWidth": 0,
        "elevation": 2,
        "shadowColor": "rgba(56, 146, 250, 0.23)",
        "shadowOffset": Object {
          "height": 5,
          "width": 5,
        },
        "shadowOpacity": 1,
        "shadowRadius": 3,
      },
    ]
  }
/>
`;

exports[`snapshot 3`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgb(56, 98, 175)",
      "borderColor": "rgba(14, 12, 29, 0.09)",
      "borderRadius": 36.14285714285714,
      "borderWidth": 1.5,
      "flexDirection": "row",
      "height": 72.28571428571428,
      "justifyContent": "space-between",
      "marginTop": 19.714285714285715,
      "opacity": 1,
    }
  }
>
  <Image
    resizeMode="contain"
    style={
      Array [
        Object {},
        Object {
          "height": 30,
          "marginLeft": 30,
          "width": 30,
        },
        Object {},
      ]
    }
  />
  <Text
    allowFontScaling={false}
    color="rgb(0, 0, 0)"
    fontFamily="SFProText-Regular"
    fontSize={24}
    style={
      Array [
        Object {
          "color": "rgb(0, 0, 0)",
          "fontFamily": "SFProText-Regular",
          "fontSize": 24,
        },
        Object {
          "color": "rgb(255, 255, 255)",
          "fontFamily": "SFProDisplay-Semibold",
          "fontSize": 25.5,
        },
        Object {},
      ]
    }
  />
  <View
    style={
      Object {
        "height": 30,
        "marginLeft": 30,
        "width": 30,
      }
    }
  >
    <Text
      allowFontScaling={false}
      color="rgb(0, 0, 0)"
      fontFamily="SFProText-Regular"
      fontSize={24}
      style={
        Array [
          Object {
            "color": "rgb(0, 0, 0)",
            "fontFamily": "SFProText-Regular",
            "fontSize": 24,
          },
          Object {
            "color": "rgb(255, 255, 255)",
            "fontFamily": "SFProDisplay-Semibold",
            "fontSize": 25.5,
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`snapshot 4`] = `
<Image
  style={
    Array [
      Object {},
    ]
  }
/>
`;

exports[`snapshot 5`] = `
<Text
  allowFontScaling={false}
  color="rgb(0, 0, 0)"
  fontFamily="SFProText-Regular"
  fontSize={24}
  style={
    Array [
      Object {
        "color": "rgb(0, 0, 0)",
        "fontFamily": "SFProText-Regular",
        "fontSize": 24,
      },
    ]
  }
/>
`;
